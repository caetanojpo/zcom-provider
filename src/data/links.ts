import WhatsAppIcon from '../../public/icons/whatsapp.svg';
import InstagramIcon from '../../public/icons/instagram.svg';
import { SVGIcon } from '@/types/svg';

type Links = {
  title?: string;
  link: string;
  target?: string;
  icon?: SVGIcon;
};

export const socialMediaLinks: Links[] = [
  {
    link: 'https://wa.me/5518996660018',
    icon: WhatsAppIcon,
    title: 'WhatsApp ZCOM',
    target: '_blank',
  },
  {
    link: 'https://www.instagram.com/zcomprovedor?igsh=MWh6eHkzYW9hb3E1ZQ%3D%3D&utm_source=qr',
    icon: InstagramIcon,
    title: 'Instagram ZCOM',
    target: '_blank',
  },
];
