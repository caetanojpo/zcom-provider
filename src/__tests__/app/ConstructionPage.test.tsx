import { mockSocialMediaLinks } from '@/__mocks__/data/links.mock';
import { render, screen } from '@testing-library/react';
import nextImageMock from '@/__mocks__/nextImage.mock';
import mockIcon from '@/__mocks__/components/atoms/icons/Icon.mock';
import { JSX } from 'react';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: Parameters<typeof nextImageMock>[0]) => nextImageMock(props),
}));

jest.mock('../../data/links.data', () => ({
  __esModule: true,
  socialMediaLinks: mockSocialMediaLinks,
}));

jest.mock('../../components/atoms/icons/Icon', () => {
  return {
    __esModule: true,
    Icon: mockIcon,
  };
});

jest.mock('../../components/atoms/backgrounds/DarkOverlay', () => {
  return {
    __esModule: true,
    default: function MockDarkOverlay(): JSX.Element {
      return <div data-testid="dark-overlay" style={{ backgroundColor: 'rgba(0,0,0,0.3)' }} />;
    },
  };
});

jest.mock('../../components/atoms/buttons/SocialIconButton', () => {
  return {
    __esModule: true,
    default: function MockSocialIconButton({
      name,
      link,
      target,
    }: {
      name?: string;
      link: string;
      target?: string;
    }): JSX.Element {
      return (
        <a data-testid={`social-${String(name)}`} href={link} target={target}>
          {name}
        </a>
      );
    },
  };
});

import ConstructionPage from '@/app/page';

describe('ConstructionPage test suite', () => {
  let container: HTMLElement;

  beforeEach(() => {
    const renderResult = render(<ConstructionPage />);
    container = renderResult.container;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should renders main heading as an h1', () => {
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(/Site em construção/i);
  });

  it('Should renders decorative background element with aria-hidden', () => {
    const decorative = container.querySelector('[aria-hidden]') as HTMLElement | null;
    expect(decorative).toBeTruthy();
    expect(decorative?.className).toEqual(expect.stringContaining('bg-construction-bg'));
  });

  it('Should renders DarkOverlay (mock) and Icon', () => {
    const overlay = screen.getByTestId('dark-overlay');
    expect(overlay).toBeInTheDocument();

    const icon = screen.getByTestId('icon');
    expect(icon).toBeInTheDocument();
  });

  it('Should renders main title and descriptive texts', () => {
    const heading = screen.getByTestId('mainText');
    const firstText = screen.getByTestId('firstText');
    const secondText = screen.getByTestId('secondText');
    const thirdText = screen.getByTestId('thirdText');

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(/Site em construção!/i);
    expect(firstText).toBeInTheDocument();
    expect(firstText).toHaveTextContent(
      /Nosso site está em construção, mas estamos prontos para começar!/i,
    );
    expect(secondText).toBeInTheDocument();
    expect(secondText).toHaveTextContent(/A Zcom está preparando algo incrível para vocês!/i);
    expect(thirdText).toBeInTheDocument();
    expect(thirdText).toHaveTextContent(
      /Clique nos ícones abaixo para acessar nossas redes sociais ou entrar em contato para saber mais./i,
    );
  });

  it('Should renders the "sob construção" image with correct src and alt', () => {
    const constructionSign = screen.getByRole('img', {
      name: /Placa "sob construção" da empresa ZCOM/i,
    });

    expect(constructionSign).toBeInTheDocument();
    expect(constructionSign).toHaveAttribute(
      'src',
      expect.stringContaining('under-construction-sign.png'),
    );
  });

  it('Should renders social media links from data and respects link count', () => {
    const allLinks = screen.getAllByRole('link');
    const socialLinks = allLinks.filter((a) =>
      mockSocialMediaLinks.some((m) => {
        return a.getAttribute('href') === m.link || a.getAttribute('href')?.includes(m.link);
      }),
    );
    const instagram = screen.getByRole('link', { name: /Instagram/i });
    const linkedin = screen.getByRole('link', { name: /LinkedIn/i });

    expect(socialLinks).toHaveLength(mockSocialMediaLinks.length);
    expect(instagram).toBeInTheDocument();
    expect(instagram).toHaveAttribute('href', expect.stringContaining('instagram.com'));
    expect(linkedin).toBeInTheDocument();
    expect(linkedin).toHaveAttribute('href', expect.stringContaining('linkedin.com'));
  });

  it('Should snapshot to detect unintentional structural changes', () => {
    expect(container).toMatchSnapshot();
  });
});
