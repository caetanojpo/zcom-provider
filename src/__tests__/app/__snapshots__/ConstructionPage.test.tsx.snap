// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ConstructionPage test suite Should snapshot to detect unintentional structural changes 1`] = `
<div>
  <main
    class="relative w-dvw h-dvh overflow-hidden text-white flex flex-col xl:flex-row items-center xl:items-start"
  >
    <div
      aria-hidden="true"
      class="absolute pointer-events-none z-10 bg-construction_bg bg-no-repeat bg-center bg-contain bottom-[-200px] left-[-100px] w-[120%] h-[420px] md:bottom-[-300px] md:left-[-200px] md:h-[600px] xl:bottom-[-500px] xl:left-[-650px] xl:h-[900px]"
    />
    <div
      class="absolute inset-0 bg-gradient-to-b xl:bg-gradient-to-br from-dark to-zcom"
    />
    <div
      data-testid="dark-overlay"
      style="background-color: rgba(0, 0, 0, 0.3);"
    />
    <section
      class="relative w-full h-[40dvh] xl:h-[70dvh] z-20 flex items-center justify-center xl:justify-start xl:order-2"
    >
      <img
        alt="Placa "sob construção" da empresa ZCOM"
        class="object-contain h-full absolute top-0 left-16 md:static md:w-auto"
        height="800"
        src="/images/construction/under-construction-sign.png"
        width="1200"
      />
    </section>
    <section
      class="w-full h-[50dvh] xl:h-dvh z-20 flex flex-col items-center justify-center gap-6 px-6 md:px-12 xl:px-32 2xl:px-64"
    >
      <div
        class="relative w-full h-fit flex items-center justify-center xl:justify-start"
      >
        <svg
          aria-hidden="true"
          class="inline-block select-none w-16 h-16 xl:w-32 xl:h-32 text-zcom hover:text-white"
          data-testid="icon"
        />
      </div>
      <div
        class="w-full max-w-3xl text-center xl:text-left"
      >
        <h1
          class="text-2xl md:text-4xl lg:text-3xl 2xl:text-4xl font-semibold tracking-tight italic"
          data-testid="mainText"
        >
          Site em 
          <span
            class="text-zcom"
          >
            construção
          </span>
          !
        </h1>
        <div
          class="mt-4 text-sm md:text-base text-white"
        >
          <p
            data-testid="firstText"
          >
            Nosso site está em construção, mas estamos prontos para começar!
          </p>
          <p
            class="mt-2"
            data-testid="secondText"
          >
            A Zcom está preparando algo incrível para vocês!
          </p>
          <p
            class="mt-10"
            data-testid="thirdText"
          >
            Clique nos ícones abaixo para acessar nossas redes sociais ou entrar em contato para saber mais.
          </p>
        </div>
        <div
          class="mt-6 flex items-center justify-center xl:justify-start gap-4"
        >
          <a
            data-testid="social-Instagram"
            href="https://instagram.com/zcom"
          >
            Instagram
          </a>
          <a
            data-testid="social-LinkedIn"
            href="https://linkedin.com/company/zcom"
          >
            LinkedIn
          </a>
        </div>
      </div>
    </section>
  </main>
</div>
`;
